test/parallel/test-dotenv-node-options.js

```diff
@@ -48,10 +48,12 @@ describe('.env supports NODE_OPTIONS', () => {
     const code = `
       require('assert')(new Date().toString().includes('Hawaii'))
     `.trim();
+    const env = { ...process.env };
+    delete env.TZ;
```

Comment:
This is wrong. This test validates if TZ is set correctly. Can you add skip: process.env.TZ !== undefined to test rather than this change?

Reply:
This deletes the `TZ` variable from the existing environment, but `TZ` is then added back in through the [`.env` file](https://github.com/philnash/node/blob/src-dont-overwrite-environment-from-dotenv/test/fixtures/dotenv/node-options.env#L4) and tested. It works because we now have the control over the env that we pass to `common.spawnPromisified`.

I can just skip it, but this seemed like a better option.

Reply:
I see. I recommend adding a comment to the changed line.

Reply:
Comment added. Now let's see all the tests pass! 🤞